### ISSUE TITLE
Please upgrade dropbox dependency from 7.x to 8 or 9

### ISSUE THREAD

**Issue by eigood** [rogebrd/dropbox-oauth-popup#142]
I'm getting a 'unable to handle fetch' type error in dropbox.sdk.min.js, which seems to be solved in 8.1.

**Comment by eigood** [comment]
Further info; I am migrating from pre-popup usage, and did not have a clientSecret.  I am told to use PKCE, which I then set that parameter, and that then leads me to the fetch error.

**Comment by eigood** [comment]
```
diff --git a/src/dropboxPopup.js b/src/dropboxPopup.js
index 11cea89..dbd1f60 100644
--- a/src/dropboxPopup.js
+++ b/src/dropboxPopup.js
@@ -73,6 +73,7 @@ export default class DropboxPopup {
     this.timeout = timeout || defaultTimeout;
 
     this.authObject = new DropboxAuth({
+      fetch: window.fetch.bind(window),
       clientId: this.clientId,
       clientSecret: this.clientSecret,
     });
@@ -110,6 +111,9 @@ export default class DropboxPopup {
        * @returns {void}
        */
       function handleRedirect(event) {
+        if (event.source !== popupWindow) {
+          return;
+        }
         window.removeEventListener('message', popup.handleRedirect);
 
         const { data } = event;
```
Basically, there could be *multiple* child windows(popups or iframes) all sending data back to the parent.  So, there needs to be a check that the source window is the one we created.  Additionally, the dropbox-sdk that is being required has a bug with fetch, that had to be worked around.

There is also an *undocumented* requirement that the redirectUri *must* be specified, and that it *must* call dispatchMessage().  This needs to be better planned out.  I implemented that myself directly in a react-router handler, and call window.opener.postMessage() inside a React.useEffect().

**Comment by rogebrd** [comment]
Hello, my apologies on how late this issue took to resolve. I have a diff up to upgrade the Dropbox dependency and have separated out the other issue mentioned to fix the multiple children and the redirectUri/call dispatch message issues here: https://github.com/rogebrd/dropbox-oauth-popup/issues/164

**Comment by rogebrd** [comment]
Hello @eigood, sorry for the delay in fixing the issue but the bug should be fixed as of version v3.1.0!
